user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/Ad_hoc_data/Ad_hoc_Building_blocks.md

```diff
@@ -14,33 +14,17 @@ All other interfaces add additional functionality.
 
 The available interfaces are:
 
-- [IGQIDataSource](xref:GQI_IGQIDataSource)
+- [IGQIDataSource](xref:GQI_IGQIDataSource): Be able to provide rows and columns.
 
   > [!IMPORTANT]
   > This is the only required interface. See also: [Configuring an ad hoc data source in a query](xref:Configuring_an_ad_hoc_data_source_in_a_query)
 
-- [IGQIInputArguments](xref:GQI_IGQIInputArguments)
+- [IGQIInputArguments](xref:GQI_IGQIInputArguments): Retrieve input from the user through input arguments.
 
-- [IGQIOnDestroy](xref:GQI_IGQIOnDestroy)
+- [IGQIOnDestroy](xref:GQI_IGQIOnDestroy): Can be implemented to clean up resources after it has been used.
 
-- [IGQIOnInit](xref:GQI_IGQIOnInit)
+- [IGQIOnInit](xref:GQI_IGQIOnInit): Can be implemented to get notified when the custom operator is created.
```

Comment:
Would write something like this instead:
Provides a way to initialize the data source with access to dependencies like e.g. the DMS